tuf/api/metadata.py

```diff
@@ -5,7 +5,21 @@
 
 This module provides container classes for TUF role metadata, including methods
 to read and write from and to file, perform TUF-compliant metadata updates, and
-create and verify signatures.
+create and verify signatures. The intent of the API is to provide safe
+serialization and deserializaiton and a safe API for accessing and modifying
+the metadata.
+
+Each of the top level metadata roles is represented by a separate class and each
+of those classes denotes exactly one metadata file.
+Additionally, there are helper classes providing abstraction over the complex
+metadata fields inside the metadata files and are used inside the four top
+level classes.
```

Comment:
This sounds good, but "four top level classes" could maybe be spelled out so it's clear what it means.